### ISSUE TITLE
Config panel values not re-applied after config auto-upgrade

### ISSUE THREAD

**Issue by Gredin67** [YunoHost-Apps/mautrix_whatsapp_ynh#99]
since auto-upgrade of config.yaml was introduced, it seems like the values from the config panel are not kept after upgrade. Which was the case when I implemented the config panel first.

**Comment by thardev** [comment]
Hi @Gredin67

How can I reproduce this? because from my point of view the autoupdater shouldn't be altering anything related to the config panel, I use the same `__VARS__` there 😥

**Comment by Gredin67** [comment]
btw you can force upgrade if you want to reproduce 
`yunohost app upgrade synapse -F -u https://github.com/Gredin67/synapse_ynh/tree/config-panel-no-sed`

**Comment by Gredin67** [comment]
with the correct url of course

**Comment by Gredin67** [comment]
Do your values get back to default after an upgrade? e.g. in relay,user,admin permissions

The issue may be on my installation 

Le 23 avril 2023 12:04:52 GMT+02:00, Dante ***@***.***> a écrit :
>Hi @Gredin67
>
>How can I reproduce this? because from my point of view the autoupdater shouldn't be altering anything related to the config panel, I use the same `__VARS__` there 😥 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/issues/99#issuecomment-1519020368
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- Sent from /e/ Mail.

**Comment by thardev** [comment]
Hi @Gredin67 

From what I've seen on my instances and during testing, values usually get restored, yes, but I'll try to do some test in my dev box to see if something is off...

**Comment by Gredin67** [comment]
The whole structure/indentation of the config file seems to have changed and I have following after upgrade :
```
An error prevents the configuration panel from being displayed:

Config panel question 'print_level' should be initialized with a value during install or upgrade.

Please report this error to the YunoHost team to get it fixed!
```

**Comment by Gredin67** [comment]
actually, config was kept at update, but there is a parameter name that changed : 
```
logging:
    min_level: info
```

**Comment by thardev** [comment]
Hi @Gredin67 

Yes, it should be fixed on the latest release actually 🙂 

https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/blob/7907662443147928dd2901c40ff3e0cf51db4f3f/.github/workflows/updater.sh#L152

EDIT: Maybe not 😅 

https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/blob/7907662443147928dd2901c40ff3e0cf51db4f3f/config_panel.toml#L196

I'll try to fix it ASAP

**Comment by thardev** [comment]
@Gredin67  Should be fixed with #106, will merge it after CI is done

**Comment by Gredin67** [comment]
fixed by #106
